fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/impl/recipe/ingredient/CustomIngredientImpl.java

```diff
@@ -150,4 +137,33 @@ public boolean isEmpty() {
 	private <T> T coerceIngredient() {
 		return (T) customIngredient;
 	}
+
+	public static <T> Codec<T> first(Codec<T> first, Codec<T> second) {
+		return new First<>(first, second);
+	}
+
+	// Decode/encode the first codec, if that fails return the result of the second.
```

Comment:
Why not `ExtraCodecs::withAlternative` (mojmap)?

Reply:
I struggled to get this to work, the encode is slightly different I think. We can always improve this later, I want to get this out now 👍